resolving/npm-resolver/src/index.ts

```diff
@@ -251,6 +258,59 @@ async function resolveNpm (
   }
 }
 
+async function tryResolveFromJsr (
+  ctx: ResolveFromNpmContext,
+  wantedDependency: WantedDependency,
+  opts: Omit<ResolveFromNpmOptions, 'registry'>,
+  defaultTag: string
+): Promise<ResolveResult | null> {
+  if (!wantedDependency.pref?.startsWith('jsr:')) return null
+
+  const registry: string | undefined = ctx.registries['@jsr']
+  if (!registry) {
+    // Q: Why throw an error instead of falling back to a default URL?
+    // A: The packages from JSR registry may have dependencies in the `@jsr` scope which would
+    //    cause confusing error messages should it not be defined.
+
+    throw new PnpmError('JSR_REGISTRY_NOT_DEFINED', 'Cannot use JSR registry without defining a registry for the `@jsr` scope', {
+      hint: 'Fix it by adding `@jsr:registry=https://npm.jsr.io` to .npmrc',
+    })
+  }
+
+  const pref = wantedDependency.pref.slice('jsr:'.length)
```

Comment:
The `jsr:` string is used in line 267. Should we store it in a variable?

Reply:
`'jsr:'` is 6 characters including the quotation marks, a variable name (that makes sense) would probably be longer. I don't see a reason to replace a string literal with a variable name considering `'jsr:'` is already self-descriptive by itself.

Reply:
That makes sense.